### ISSUE TITLE
Alert for HA Pair Going Down

### ISSUE THREAD

**Issue by johnwarlick** [NetApp/harvest#2315]
A customer has faced a couple of instances where an HA pair has gone down abruptly, and wanted to know the best metric(s)/alert(s) to use to quickly be on top of any such future scenarios. They would prefer to not rely on AIQUM. 

Based on [feedback in Discord](https://discord.com/channels/855068651522490400/1138531100839379115/1138531100839379115), it seems the [EMS Collector](https://netapp.github.io/harvest/nightly/configure-ems/) is the best way to go. I have found the following EMS alerts: 

- callhome.hainterconnect.down
- ic.HAInterconnectDown
- ic.HAInterconnectLinkDown
- ic.linkStatusChange
- cf.ic.heartBeatFailed

A combination of one or more these and perhaps some other EMS event(s) could be used to create an an HA pair down alert. I will pick back up on working on this is a lab on demand.

**Comment by cgrinds** [comment]
Paqui on [Discord](https://discord.com/channels/855068651522490400/1174011885654392954/1174011885654392954) has a similar request.

**Comment by johnwarlick** [comment]
We have been using the Health dashboard from 23.05.0 to check for nodes going down. When combined with the approach you outlined, I believe it combines to mostly solve it. The edge case I am slightly unsure on is if an HA pair goes down suddenly and simultaneously, i.e. a datacenter mishap. I believe the ha.takeover_check.takeover_possible should still return false, and the health dashboard should still show node down, correct?

**Comment by faguayot** [comment]
@rahulguptajss I have deployed the version `harvest version 23.12.11-nightly (commit 006cc7f0) (build date 2023-12-11T00:23:26-0500) linux/amd64`  of this night but I don't have the metric `health_ha_alerts`. 

![image](https://github.com/NetApp/harvest/assets/85484449/fbbdbcfa-82ca-459c-9c4f-de52aac74aec)

**Comment by faguayot** [comment]
@rahulguptajss So it only has a failed status? I mean it isn't a constant check that says the node/s are OK, right? Only when a problem happens is when there is any state.

**Comment by cgrinds** [comment]
Verified in `24.02` 84e2f7ca

`health_ha_alerts` metric is published

health_ha_alerts{cluster="umeng-aff300-05-06", datacenter="nane", instance="localhost:12991", job="prometheus", node="umeng-aff300-05", partner="umeng-aff300-06", partner_state="Up", severity="error", state_description="Connected to umeng-aff300-06, Takeover is not possible: Storage failover is disabled", takeover_possible="false"}
	1
health_ha_alerts{cluster="umeng-aff300-05-06", datacenter="nane", instance="localhost:12991", job="prometheus", node="umeng-aff300-06", partner="umeng-aff300-05", partner_state="Up", severity="error", state_description="Connected to umeng-aff300-05, Takeover is not possible: Storage failover is disabled", takeover_possible="false"}
 

<img width="439" alt="image" src="https://github.com/NetApp/harvest/assets/242252/9dcf31fa-130e-4f45-8777-23ff2857fa44">

**Comment by faguayot** [comment]
Hello, 
We finally could test this new information but the thing is that we can't use it for alerting since there the value for alerts don't gather any information when the node recover its health. You can take a look on the information gathered for both databases that we use InfluxDB and Prometheus but we must use InfluxDB for alerting. 

Below the metrics `health_ha_alert`s and `health_node_alerts` in **Prometheus** and `health_ha` and `health_nod`e in **InfluxDB**

- InfluxDB
![image](https://github.com/NetApp/harvest/assets/85484449/e50854a0-59ab-4270-98cd-dd7632a1f373)
![image](https://github.com/NetApp/harvest/assets/85484449/d74ebd5a-d79d-459a-9be5-5362a4b0bcda)


- Prometheus
![image](https://github.com/NetApp/harvest/assets/85484449/70e65c00-0d78-4760-ae68-447b80df7ff9)
![image](https://github.com/NetApp/harvest/assets/85484449/8f44891d-a09d-4ed9-a74d-e3006b06feca)

In summary if no value is inserted when retrieving at least 1 time, we have no way of comparing what we were receiving to perform the retrieval of the same.

Thanks.

**Comment by faguayot** [comment]
On InfluxDB, at least how our monitoring team have deployed the stack in the organization, we need a value to recover an alert. If we don't receive any other information, the alert is still happening. So we don't have an automatic recovery after some time.

@rahulguptajss Yes, I can open a request for this.

**Comment by faguayot** [comment]
@rahulguptajss Apologize por delay, I didn't have time to check this and open the ticket for the new feature. Thanks so much for creating.
